tests/Unit/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/VerifyEinsteinSolution.cpp

```diff
@@ -0,0 +1,284 @@
+                                g.get(k, j) * d_shift.get(i, k);
+        }
+      }
+    }
+    for (size_t i = 0; i < 3; ++i) {
+      for (size_t j = 0; j < 3; ++j) {
+        CHECK_ITERABLE_APPROX(dt_g_adm.get(i, j), zero);
+      }
+    }
+  }
+
+  // Test 1-index constraint
+  auto C_1 = make_with_value<tnsr::a<DataVector, 3>>(x, 0.0);
+  for (size_t a = 0; a < 4; ++a) {
+    C_1.get(a) = gauge_function.get(a);
+    for (size_t i = 0; i < 3; ++i) {
+      for (size_t j = 0; j < 3; ++j) {
+        C_1.get(a) += upper_spatial_metric.get(i, j) * phi.get(i, j + 1, a);
+      }
+    }
+    for (size_t b = 0; b < 4; ++b) {
+      C_1.get(a) += normal_vector.get(b) * pi.get(b, a);
+      for (size_t c = 0; c < 4; ++c) {
+        C_1.get(a) -=
+            0.5 * normal_one_form.get(a) * pi.get(b, c) * upper_psi.get(b, c);
+        if (a > 0) {
+          C_1.get(a) -= 0.5 * phi.get(a - 1, b, c) * upper_psi.get(b, c);
+        }
+        for (size_t i = 0; i < 3; ++i) {
+          C_1.get(a) -= 0.5 * normal_one_form.get(a) *
+                        normal_vector.get(i + 1) * phi.get(i, b, c) *
+                        upper_psi.get(b, c);
+        }
+      }
+    }
+  }
+  for (size_t a = 0; a < 4; ++a) {
+    for (size_t s = 0; s < data_size; ++s) {
+      CHECK(C_1.get(a)[s] == numerical_approx(0.0));
+    }
+  }
+
+  // Constraint-damping parameters: Set to arbitrary values.
+  // gamma1 = 0 and -1 are special, so don't choose that one.
+  const auto gamma0 = make_with_value<Scalar<DataVector>>(x, 1.0);
+  const auto gamma1 = make_with_value<Scalar<DataVector>>(x, 1.0);
+  const auto gamma2 = make_with_value<Scalar<DataVector>>(x, 1.0);
+
+  // Compute RHS of generalized harmonic Einstein equations.
+  auto dt_psi =
+      make_with_value<tnsr::aa<DataVector, 3, Frame::Inertial>>(x, 0.0);
+  auto dt_pi =
+      make_with_value<tnsr::aa<DataVector, 3, Frame::Inertial>>(x, 0.0);
+  auto dt_phi =
+      make_with_value<tnsr::iaa<DataVector, 3, Frame::Inertial>>(x, 0.0);
+  GeneralizedHarmonic::ComputeDuDt<3>::apply(
+      make_not_null(&dt_psi), make_not_null(&dt_pi), make_not_null(&dt_phi),
+      psi, pi, phi, d_psi, d_pi, d_phi, gamma0, gamma1, gamma2, gauge_function,
+      d4_H, lapse, shift, upper_spatial_metric, upper_psi,
+      trace_christoffel_first_kind, christoffel_first_kind,
+      christoffel_second_kind, normal_vector, normal_one_form);
+
+  // Make sure the RHS is zero.
+  for (size_t a = 0; a < 4; ++a) {
+    for (size_t b = 0; b < 4; ++b) {
+      for (size_t s = 0; s < data_size; ++s) {
+        CHECK(dt_psi.get(a, b)[s] == numerical_approx(0.0));
+        CHECK(dt_pi.get(a, b)[s] == numerical_approx(0.0));
+      }
+      for (size_t i = 0; i < 3; ++i) {
+        for (size_t s = 0; s < data_size; ++s) {
+          CHECK(dt_phi.get(i, a, b)[s] == numerical_approx(0.0));
+        }
+      }
+    }
+  }
+}
+
+// Need explicit instantiation of derivatives.
+
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.cpp"
```

Comment:
Why is this include needed? If this is really the way to do this then I'd add another line after it.

Reply:
This is how it is designed to be used, but actually looking at this I think it's kind of an invitation to have everything blow up in our faces.  In particular, if the included file contains anything that isn't templated (or inline) we'll likely get linking errors.

PartialDerivative.cpp needs to follow all the rules for header files, and I think we should rename it to something ending in hpp to avoid confusion.